src/ORM/Behavior/TimestampBehavior.php

```diff
@@ -136,11 +136,11 @@ public function implementedEvents(): array
      * If an explicit date time is passed, the config option `refreshTimestamp` is
      * automatically set to false.
      *
-     * @param \DateTime|null $ts Timestamp
+     * @param \DateTimeInterface|null $ts Timestamp
      * @param bool $refreshTimestamp If true timestamp is refreshed.
-     * @return \DateTime
+     * @return \DateTimeInterface
      */
-    public function timestamp(?DateTime $ts = null, bool $refreshTimestamp = false): DateTime
+    public function timestamp(?DateTimeInterface $ts = null, bool $refreshTimestamp = false): DateTimeInterface
```

Comment:
As signature change this can be BC breaking and should also be noted in the migration guide IMO.
Extending classes would be in trouble.

Reply:
Huh? All the typehint additions are already going to require changing extending classes. Why is this any different which requires special mention?

Reply:
Why would subclasses be in trouble? They are getting an interface that allows immutable types and mutable ones which is a wider type definition.